gazebo_plugins/include/gazebo_plugins/gazebo_ros_depth_camera.h

```diff
@@ -138,6 +138,7 @@ namespace gazebo
     private: int depth_info_connect_count_;
     private: void DepthInfoConnect();
     private: void DepthInfoDisconnect();
+    private: bool use_depth_image_16UC1_format_;
```

Comment:
Adding a private member variable will break ABI, so this would have to target Noetic :confused:

Reply:
Given the nature of ROS we don't need to keep ABI stable since the buildfarm will recompile all the dependencies on top. For `gazebo_ros_pkgs` we need to keep API, topic/services layout and behavior but I think that we can safely break the ABI.

Reply:
Ok, I misunderstood it then. I thought that we were trying to keep ABI stable for the packages which are not on the ROS build farm, like the ones on http://packages.osrfoundation.org/ and other custom packages that are built downstream on top of ours.

Reply:
not breaking the ABI is a nice to have but I think that in the ROS1 world it should not be a limitation and external packages needs to be adapted (recompiled) according.

Reply:
Soo... Can we merge then?